Source/CNTKv2LibraryDll/PrimitiveFunction.h

```diff
@@ -713,8 +714,8 @@ namespace CNTK
         static void FixNDShape(size_t filterRank, size_t inputRank, NDShape& shape, size_t deflt, const NDShape& from = NDShape());
 
         static NDShape ConvolutionOpOutputShape(PrimitiveOpType op, const NDShape& operandShape, NDShape& kernelShape, NDShape& outputMapCount, NDShape& strides,
-            std::vector<bool>& sharing, std::vector<bool>& autoPad, NDShape& lowerPad, NDShape& upperPad,
-            bool transpose, bool inferDimensions, bool ceilOutputDim = false);
+                                                std::vector<bool>& sharing, std::vector<bool>& autoPad, NDShape& lowerPad, NDShape& upperPad,
+                                                bool transpose, bool inferDimensions, NDShape dilation = NDShape({1}), bool ceilOutputDim = false);
```

Comment:
NDShape dilation should be by reference, NDShape& dilation.

Reply:
Also FixNDShape(fillRank, inputRank, dilation, 1); instead of FixNDShape(fillRank, inputRank, dilation, 1, fromShape);. The latter will add non 1 at the end.